Helper/Discount.php

```diff
@@ -49,6 +49,9 @@ class Discount extends AbstractHelper
     const MAGEPLAZA_GIFTCARD = 'gift_card';
     const MAGEPLAZA_GIFTCARD_QUOTE_KEY = 'mp_gift_cards';
     const AHEADWORKS_STORE_CREDIT = 'aw_store_credit';
+    const BSS_STORE_CREDIT = 'bss_storecredit';
+    const AMASTY_COUPON_AMOUNT = 'amasty_coupon_amount';
+    const AHEADWORKS_GIFTCARD = 'aw_giftcard';
```

Comment:
it seems like previously we already added `AHEADWORKS_STORE_CREDIT` is there anyway we can find all `aheadworks` discounts and add them now? same for amasty. we should just be able to whitelist another module once and for all as a whole?

Reply:
@roopakv those are different modules of the same vendor. Support for a module can be a not so trivial task and so far it was done on demand (when a merchant needs it we build it into our plugin as an integral part).
@oldDriver did you test the case I explained above with all of the newly added discounts?

Reply:
@danac-gs thank you for explaining

Reply:
@roopakv as for me better way to change property $discountTypes from "private" to "protected"
I this way we will not need to add all gift/discounts/coupons/store credit/voucher in the Helper\Cart.
We could simply add Botpay\Custom\Helper\Cart:
namespace Bolt\Custom\Helper
use Bolt\Boltpay\Helper\Cart as BaseCart;
class Cart extends BaseCart
{
    protected $discountTypes = [
        // Types we need in the current integration
    ];
}
Does it make sense?

Reply:
sort of. not following a 100% but i think i get the idea